internal/pkg/cli/task_run.go

```diff
@@ -138,7 +154,59 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute deploys and runs the task.
 func (o *runTaskOpts) Execute() error {
+	if err := o.deployTaskResources(); err != nil {
+		return err
+	}
+	// NOTE: if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		if err := o.buildAndPushImage(); err != nil {
+			return err
+		}
+
+		if err := o.updateTaskResources(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) deployTaskResources() error {
+	o.spinner.Start(fmt.Sprintf("Deploying resources for task %s", color.HighlightUserInput(o.groupName)))
+	if err := o.deploy(); err != nil {
+		o.spinner.Stop(log.Serrorln("Failed to deploy task resources."))
+		return fmt.Errorf("failed to deploy resources for task %s: %w", o.groupName, err)
+	}
+	o.spinner.Stop(log.Ssuccessln("Successfully deployed task resources."))
+	return nil
+}
+
+func (o *runTaskOpts) updateTaskResources() error {
+	o.spinner.Start(fmt.Sprintf("Updating image to task %s", color.HighlightUserInput(o.groupName)))
+	if err := o.deploy(); err != nil {
+		o.spinner.Stop(log.Serrorln("Failed to update task resources."))
+		return fmt.Errorf("failed to update resources for task %s: %w", o.groupName, err)
```

Comment:
nit: for errors we try to avoid words like "failed to": https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

> When adding context to returned errors, keep the context succinct by avoiding phrases like "failed to", which state the obvious and pile up as the error percolates up through the stack:

instead:
```suggestion
		return fmt.Errorf("update resources for task %s: %w", o.groupName, err)
```